content/docs/user-guide/experiment-management/running-experiments.md

```diff
@@ -0,0 +1,569 @@
+      torch.save(model.state_dict(), "model.pt")
+      # Evaluate and checkpoint.
+      evaluate(model, x_test, y_test)
+      dvc.api.make_checkpoint()
+    ...
+```
+
+So, even if you're not using one of these libraries, you can use checkpoints in
+your project at each epoch/step by first recording all intermediate artifacts
+and metrics, then calling `dvc.api.make_checkpoint()`.
+
+#### Adding Checkpoints to Non-Python Code
+
+If your project is written in another language, you can mimic the behavior of
+`make_checkpoint` in your project. In essence `make_checkpoint` creates a
+special file named `DVC_CHECKPOINT` inside `.dvc/tmp/` to signal the checkpoint,
+and waits DVC to delete it.
+
+```r
+
+dvcroot <- Sys.getenv("DVC_ROOT")
+
+if (dvcroot != "") {
+    signalfilepath = file.path(dvcroot, ".dvc", "tmp", "DVC_CHECKPOINT")
+    file.create(signalfilepath)
+    while (file.exists(signalfilepath)) {
+      Sys.sleep(0.01)
+    }
+
+}
+
+```
+
+The following Julia snippet creates a signal file to create a checkpoint.
+
+```julia
+
+dvc_root =  get(ENV, "DVC_ROOT", "")
+
+if dvc_root != ""
+   signal_file_path = joinpath(dvc_root, ".dvc", "tmp", "DVC_CHECKPOINT")
+   open(signal_file_path, "w") do io
+           write(io, "")
+   end;
+   while isfile(signal_file_path)
+        sleep()
+   end;
+```
+
+### Running the Experiments with Checkpoints
+
+Running the experiments with checkpoints is no different than running the
+experiments pipeline.
+
+```dvc
+$ dvc exp run -S param=value
+```
+
+All checkpoints registered at runtime will be preserved, even if the process
+gets interrupted (e.g. with `Ctrl+C`, or by an error). Without interruption, a
+"wrap-up" checkpoint will be added (if needed), so that changes to pipeline
+outputs don't remain in the workspace.
+
+Subsequent uses of `dvc exp run` will continue from the latest checkpoint (using
+the latest cached versions of all outputs).
+
+You can list previous checkpoints with `dvc exp show`. To resume from a previous
+checkpoint, you must first `dvc exp apply` it before using `dvc exp run`. For
+`--queue` or `--temp` runs, use `--rev` to specify the checkpoint to continue
+from.
+
+Alternatively, use `--reset` to start over (discards previous checkpoints and
+their outputs). This is useful for re-training ML models, for example.
+
+> Note that queuing an experiment that uses checkpoints implies `--reset`,
+> unless a `--rev` is provided (refer to the previous section).
+
+<details>
+
+### How are checkpoints captured?
+
+Instead of a single commit, checkpoint experiments have multiple commits under
+the custom Git reference (in `.git/refs/exps`), similar to a branch.
+
+</details>
+
+## Git and Experiments
+
+Experiments are custom
+[Git references](https://git-scm.com/book/en/v2/Git-Internals-Git-References)
+(found in `.git/refs/exps`) with a single commit based on `HEAD` (not checked
+out by DVC). Note that these commits are not pushed to the Git remote by default
```

Comment:
All of this should probably be in the section index (https://dvc.org/doc/user-guide/experiment-management) or maybe it could work as the document's short intro/background.

Reply:
Or as PART of the intro, but as a hidden section (a bit too technical).

Reply:
It may be separated into a section by adding some content from https://dvc.org/blog/experiment-refs I believe. It's a bit alien to this section, you're correct.

Reply:
> adding some content from https://dvc.org/blog/experiment-refs

The reason it was written as a blog was to avoid those implementation details here 😅

Reply:
Why do we put extra-technical details in a blog?

Reply:
Removing this section from here. Maybe we can tell elsewhere.